zeebe/atomix/cluster/src/main/java/io/atomix/cluster/AtomixCluster.java

```diff
@@ -118,19 +113,15 @@ protected AtomixCluster(
       final ManagedMessagingService messagingService,
       final ManagedUnicastService unicastService,
       final String actorSchedulerName) {
-
+    config.getProtocolConfig().setActorSchedulerName(actorSchedulerName);
```

Comment:
:thought_balloon:  It is not really a configuration for the protocol. I was thinking of passing it directly to the constructor the membership protocol. So in `buildMembershipProtocol`:

```
config.getProtocolConfig().getType().newProtocol(config.getProtocolConfig(), actorSchedulerName);
```

What do you think?

Reply:
>  I had tried as you suggested but that introduces more changes due to the modification of GroupMembershipProtocol.newProtocol 

Is that an issue? The only main change is in the interface `newProtocol` and the two implementations of it, right? 

`buildClusterMembership` and `buildMessagingService` are internal methods in `AtomixCluster`. So I guess we are not getting much benefit by removing the parameter from there.

Reply:
I agree with that view since those methods are internal.